src/api/spec/models/comment_spec.rb

```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+RSpec.shared_examples "a comment" do
+  let(:comment_factory) { described_class.name.underscore }
+  let(:comment) { create(comment_factory) }
+
+  describe "A comment" do
+    it { expect(build(comment_factory)).to be_valid }
+  end
+
+  describe "Comment associations" do
+    it { is_expected.to belong_to(:bs_request).inverse_of(:comments) }
+    it { is_expected.to belong_to(:project).inverse_of(:comments) }
+    it { is_expected.to belong_to(:package).inverse_of(:comments) }
+    it { is_expected.to belong_to(:user).inverse_of(:comments) }
+
+    it { is_expected.to have_many(:children).dependent(:destroy).class_name('Comment').with_foreign_key('parent_id') }
+  end
+
+  describe "Comment validations" do
+    it { is_expected.to validate_presence_of(:body) }
+    it { is_expected.to validate_presence_of(:type) }
+    it { is_expected.to validate_presence_of(:user) }
+  end
+
+  describe "to_xml" do
+    let(:builder) { builder = Nokogiri::XML::Builder.new }
+
+    context "without parent" do
+      before do
+        comment.to_xml(builder)
+      end
+
+      context "returns xml" do
+        context "comment element" do
+          let(:comment_element) { builder.doc.css('comment') }
```

Comment:
`let(:comment_element) { builder.doc.css('comment') }` is also several time, please move it up so you don't need to have it twice.